### ISSUE TITLE
Authentication not working

### ISSUE THREAD

**Issue by supabibz** [loft-sh/vcluster#1035]
### What happened?

I created a vcluster-k8s using helm chart in ArgoCD. Everything is working fine (no error in pods).
I retrieved the kubeconfig as described in the doc  :

`kubectl get secret vc-vcluster -n vcluster --template={{.data.config}} | base64 -d > /home/XXXXX/.kube/vcluster-config`

But I get errors when trying to  get namespaces :

```
kubectl --kubeconfig /home/XXXXX/.kube/vcluster-config --insecure-skip-tls-verify=true
E0512 18:23:06.597407   51662 memcache.go:265] couldn't get current server API group list: unknown
E0512 18:23:06.639217   51662 memcache.go:265] couldn't get current server API group list: unknown
E0512 18:23:06.680011   51662 memcache.go:265] couldn't get current server API group list: unknown
E0512 18:23:06.718425   51662 memcache.go:265] couldn't get current server API group list: unknown
E0512 18:23:06.747084   51662 memcache.go:265] couldn't get current server API group list: unknown
Error from server (Forbidden): unknown
```

With more verbosity :

```
I0512 17:58:13.170245   32501 loader.go:373] Config loaded from file:  /home/XXXX/.kube/vcluster-config
I0512 17:58:13.170876   32501 round_trippers.go:466] curl -v -XGET  -H "Accept: application/json;g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList,application/json" -H "User-Agent: kubectl/v1.27.1 (linux/amd64) kubernetes/4c94112" 'https://vcluster.k8s.XXXXX/api?timeout=32s'
I0512 17:58:13.171892   32501 round_trippers.go:495] HTTP Trace: DNS Lookup for vcluster.XXXXXX resolved to [{X.X.X.X }]
I0512 17:58:13.172362   32501 round_trippers.go:510] HTTP Trace: Dial to tcp:X.X.X.X:443 succeed
I0512 17:58:13.213979   32501 round_trippers.go:553] GET https://vcluster.XXXXXX/api?timeout=32s 403 Forbidden in 43 milliseconds
I0512 17:58:13.214000   32501 round_trippers.go:570] HTTP Statistics: DNSLookup 0 ms Dial 0 ms TLSHandshake 3 ms ServerProcessing 37 ms Duration 43 ms
I0512 17:58:13.214010   32501 round_trippers.go:577] Response Headers:
I0512 17:58:13.214022   32501 round_trippers.go:580]     Audit-Id: cb2a5bfd-e17c-4c63-9eee-2ebd090d038f
I0512 17:58:13.214030   32501 round_trippers.go:580]     Audit-Id: 0343164c-238b-487e-95c5-5e5345ac761c
I0512 17:58:13.214040   32501 round_trippers.go:580]     Cache-Control: no-cache, private
I0512 17:58:13.214049   32501 round_trippers.go:580]     Cache-Control: no-cache, private
I0512 17:58:13.214058   32501 round_trippers.go:580]     X-Content-Type-Options: nosniff
I0512 17:58:13.214067   32501 round_trippers.go:580]     X-Kubernetes-Pf-Flowschema-Uid: 13bc38f2-4db8-4fd1-b16e-0388e6de537b
I0512 17:58:13.214077   32501 round_trippers.go:580]     X-Kubernetes-Pf-Prioritylevel-Uid: 98eb7deb-156a-4f6f-9d0e-8f58a696aef1
I0512 17:58:13.214086   32501 round_trippers.go:580]     Date: Fri, 12 May 2023 15:58:13 GMT
I0512 17:58:13.214094   32501 round_trippers.go:580]     Content-Type: application/json
I0512 17:58:13.214103   32501 round_trippers.go:580]     Content-Length: 188
I0512 17:58:13.214112   32501 round_trippers.go:580]     Strict-Transport-Security: max-age=15724800; includeSubDomains
I0512 17:58:13.214339   32501 request.go:1188] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"forbidden: User \"system:anonymous\" cannot get path \"/api\"","reason":"Forbidden","details":{},"code":403}
```

### What did you expect to happen?

Retrieved the vcluster namespaces list

### How can we reproduce it (as minimally and precisely as possible)?

Install vcluster-k8s helm chart 0.15 using ArgoCD

### Anything else we need to know?

*No response*

### Host cluster Kubernetes version

<details>

```
$ kubectl version
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.1", GitCommit:"4c9411232e10168d7b050c49a1b59f6df9d7ea4b", GitTreeState:"clean", BuildDate:"2023-04-14T13:14:42Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Host cluster Kubernetes distribution

<details>

```
k8s vanilla
```

</details>

### vlcuster version

<details>

```
NOT USING vcluster CLIENT
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k8s
```

</details>

### OS and Arch

<details>

```
OS:  Ubuntu 20.04
Arch: x86_64
```

</details>

**Comment by supabibz** [comment]
FYI, the content of the kube (removed sensitive data of course)

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: ...
    server: https://vcluster.XXXXXX
  name: my-vcluster
contexts:
- context:
    cluster: my-vcluster
    user: my-vcluster
  name: my-vcluster
current-context: my-vcluster
kind: Config
preferences: {}
users:
- name: my-vcluster
  user:
    client-certificate-data: ...
    client-key-data: ...
```

**Comment by ishankhare07** [comment]
Hi @supabibz , thanks for creating this issue. Have you tried looking at a few arguments to the create command like
`--out-kube-config-server`. Ref: [config-reference](https://www.vcluster.com/docs/config-reference)

> I retrieved the kubeconfig as described in the doc  
Also, can you point to which doc you're referring to here?

**Comment by mpetason** [comment]
So the issue was related to the output of the command in this doc (https://www.vcluster.com/docs/using-vclusters/kube-context#retrieving-the-kube-config-from-the-vcluster-secret) usually just defaults to my-vcluster and doesn't match the output you would expect, however if you update the config yaml for the vCluster (along the same lines as what @ishankhare07 is saying) you can then get the config to be set up correctly. The only thing that is assumed knowledge is how the kubecontext name will be configured by vCluster. 

Basically you have to know what the context name should be, which will update the configuration so that it matches the same output you would get if you were using the vCluster CLI.

In this example my vCluster is in the demo-vcluster namespace and is named demo-vcluster.

```
syncer:
  extraArgs:
  - --tls-san=demo.vcluster.local
  - --out-kube-config-server=https://demo.vcluster.local
  - --kube-config-context-name=vcluster_demo-vcluster_demo-vcluster_default
```
